pkg/policy/api/selector.go

```diff
@@ -232,19 +260,14 @@ func (n *EndpointSelector) AddMatch(key, value string) {
 		n.MatchLabels = map[string]string{}
 	}
 	n.MatchLabels[key] = value
+	n.requirements = labelSelectorToRequirements(n.LabelSelector)
 }
 
 // Matches returns true if the endpoint selector Matches the `lblsToMatch`.
 // Returns always true if the endpoint selector contains the reserved label for
 // "all".
 func (n *EndpointSelector) Matches(lblsToMatch k8sLbls.Labels) bool {
-	lbSelector, err := metav1.LabelSelectorAsSelector(n.LabelSelector)
-	if err != nil {
-		// FIXME: Omit this error or throw it to the caller?
-		// We are doing the verification in the ParseEndpointSelector but
-		// don't make sure the user can modify the current labels.
-		log.WithError(err).WithField(logfields.EndpointLabelSelector,
-			logfields.Repr(n)).Error("unable to match label selector in selector")
+	if n.requirements == nil {
 		return false
```

Comment:
I think we should fall back to the original matching algorithm if we can't use the cached requirements.

Reply:
Makes sense, will fix.

Reply:
I fall back to using the original algorithm if the requirements are nil.